### PR TITLE
Pull Falcon updates from PQClean

### LINKED ISSUE #1488: 0.8.0rc1 MSVC 32-bit builds fail on Falcon with AVX
### Discussed in https://github.com/orgs/open-quantum-safe/discussions/1487

<div type='discussions-op-text'>

<sup>Originally posted by **blackbird1999** May 31, 2023</sup>
 44>D:\CDK8.1\liboqs\src\sig\falcon\pqclean_falcon-512_avx2\sign.c(1033): error C2440: 'function': cannot convert fr
       om 'int' to '__m128i' [D:\CDK8.1\liboqs\buildmt\src\sig\falcon\falcon_512_avx2.vcxproj]
    41>D:\CDK8.1\liboqs\src\sig\falcon\pqclean_falcon-1024_avx2\sign.c(1033): error C2440: 'function': cannot convert f
       rom 'int' to '__m128i' [D:\CDK8.1\liboqs\buildmt\src\sig\falcon\falcon_1024_avx2.vcxproj]

It appears that the AVX2 code no longer matches that found here https://falcon-sign.info/falcon-round3.zip. In particular, it no longer has the #ifdef for x64/x86 that allows it to compile. 

I'm not sure of the steps to report such things. Thanks for your help!

Sincerely,
Jonathan
</div>

### PR DESCRIPTION
**Pull request by dstebila**
https://github.com/PQClean/PQClean/commit/c3abebf4ab1ff516ffa71e6337f06d898952c299

Fixes #1488.

* [No] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [No] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in [oqs-provider](https://github.com/open-quantum-safe/oqs-provider), [OQS-OpenSSL](https://github.com/open-quantum-safe/openssl), [OQS-BoringSSL](https://github.com/open-quantum-safe/boringssl), and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: docs/algorithms/kem/kyber.yml [hunk 1]
```diff
@@ -22,9 +22,10 @@ primary-upstream:
   spdx-license-identifier: CC0-1.0 or Apache-2.0
 optimized-upstreams:
   pqclean-aarch64:
-    source: https://github.com/PQClean/PQClean/commit/66e50172055aaf1b9a16d8f35fe03b0807f2723e
+    source: https://github.com/PQClean/PQClean/commit/c3abebf4ab1ff516ffa71e6337f06d898952c299
       with copy_from_upstream patches
-    spdx-license-identifier: CC0-1.0 and (CC0-1.0 or Apache-2.0) and (CC0-1.0 or MIT) and MIT
+    spdx-license-identifier: CC0-1.0 and (CC0-1.0 or Apache-2.0) and (CC0-1.0 or MIT)
+      and MIT
```
Comment (SWilson4) [inline docs/algorithms/kem/kyber.yml:28]:
```suggestion
    spdx-license-identifier: CC0-1.0 and (CC0-1.0 or Apache-2.0) and (CC0-1.0 or MIT) and MIT
```

Comment (SWilson4) [reply docs/algorithms/kem/kyber.yml:28]: Looks like an extra newline sneaked in here (unintentionally?)

File: docs/algorithms/sig/dilithium.yml [hunk 2]
```diff
@@ -20,9 +20,10 @@ primary-upstream:
   spdx-license-identifier: CC0-1.0 or Apache-2.0
 optimized-upstreams:
   pqclean-aarch64:
-    source: https://github.com/PQClean/PQClean/commit/66e50172055aaf1b9a16d8f35fe03b0807f2723e
+    source: https://github.com/PQClean/PQClean/commit/c3abebf4ab1ff516ffa71e6337f06d898952c299
       with copy_from_upstream patches
-    spdx-license-identifier: CC0-1.0 and (CC0-1.0 or Apache-2.0) and (CC0-1.0 or MIT) and MIT
+    spdx-license-identifier: CC0-1.0 and (CC0-1.0 or Apache-2.0) and (CC0-1.0 or MIT)
+      and MIT
```
[the remark this answers is not in the document]
Comment (SWilson4) [reply docs/algorithms/sig/dilithium.yml:26]: Same as above.

**Review by SWilson4 — APPROVED** [review]
Looks good to me.

**Comment by dstebila** [comment]
I'm not sure what aspect of the code generation script caused the line breaks -- can you take a quick look @SWilson4?

**Comment by SWilson4** [comment]
> I'm not sure what aspect of the code generation script caused the line breaks -- can you take a quick look @SWilson4?

It turns out that it's expected behaviour of the `yaml.dump` function, which breaks lines at some default width unless an optional `width` key is provided:

https://github.com/open-quantum-safe/liboqs/blob/be678118a589af9a28b57aec7ea2609d6d0fa53d/scripts/copy_from_upstream/update_upstream_alg_docs.py#L23-L25

Apparently the result is still valid YAML, so I don't think we need to worry about it.

**Pull request merged**